main.js

```diff
@@ -0,0 +1,145 @@
+const form = document.getElementById("form");
+const text = document.getElementById("text"); // 入力されたテキスト
+const addButton = document.getElementById("addButton");
+const taskList = document.getElementById("taskList");
+const radioForm = document.getElementById("radioForm");
+let taskArray = [
+  // {
+  //   id: 0,
+  //   name: "ほげ",
+  //   status: "作業中",
+  // },
+  // {
+  //   id: 1,
+  //   name: "ほげ",
+  //   status: "作業中",
+  // },
+  // {
+  //   id: 2,
+  //   name: "ほげ",
+  //   status: "作業中",
+  // },
+];
+
+const displayTaskArray = (taskArray) => {
+  console.log("@@@", taskArray);
+  // tbodyを初期化
+  taskList.innerHTML = "";
+
+  taskArray.forEach((task) => {
+    // 配列の中のオブジェクトの数だけ処理を繰り返す
+    const tr = document.createElement("tr");
+    taskList.appendChild(tr);
+    const objArray = Object.entries(task); // オブジェクトを配列に変換
+    // console.log(objArray);
+    objArray.forEach((arr) => {
+      const td = document.createElement("td");
+      const buttonTagStatus = document.createElement("button");
+      const buttonTagDelete = document.createElement("button");
+      if (arr[0] === "status") {
+        buttonTagStatus.innerHTML = task.status;
+        td.appendChild(buttonTagStatus);
+        tr.appendChild(td);
+      } else if (arr[0] === "delete") {
+        buttonTagDelete.innerHTML = "削除";
+        td.appendChild(buttonTagDelete);
+        tr.appendChild(td);
+      } else {
+        td.textContent = arr[1]; // arr[1]はvalueの部分
+        tr.appendChild(td);
+      }
+
+      buttonTagStatus.addEventListener(
+        "click",
+        () => {
+          toggleStatus(task.id);
+        },
+        false
+      );
+
+      buttonTagDelete.addEventListener(
+        "click",
+        () => {
+          deleteTask(task.id);
+        },
+        false
+      );
+    });
+  });
+};
+
+displayTaskArray(taskArray);
+
+const addTaskArray = (taskArray) => {
+  taskArray.push({
+    id: taskArray.length,
+    name: text.value,
+    status: "作業中",
+    delete: "削除",
+  });
+};
+
+const toggleStatus = (id) => {
+  taskArray = taskArray.map((task) => {
+    if (task.id === id) {
+      return {
+        ...task,
+        status: task.status === "作業中" ? "完了" : "作業中",
+      };
+    } else {
+      return task;
+    }
```

Comment:
こっちのほうがスッキリとしたコードに見えます

```suggestion
    }
    return task;
```